### DISCUSSION TITLE
Question: How to skip folders which exist only on one side?

### DISCUSSION THREAD

**Discussion by CharlesMengCA** [WinMerge/winmerge#2028] [General]
Hi,

For example,  Folder A has two sub folders 

- A1
- A2

Folder B has only one sub folder

- A2

When WinMerge is comparing Folder A and Folder B, even folder A1 only exists on one side, WinMerge seems to check every file in A1. When A1 has lots of files, this will take a lot time.

How can I avoid this?

Thanks,

**Comment by sdottaka** [comment]
Disabling "Include unique subfolders contents" in the Compare/Folders category of the Options window will skip the comparison for folders that exist on only one side.

![image](https://github.com/WinMerge/winmerge/assets/98126/c9a3488b-6d86-4e2a-87fe-57a3a329cce5)

**Comment by CharlesMengCA** [comment]
Disabling "Include unique subfolders contents" should be the default option IMHO. Otherwise, some new users may think WinMerge is slow and stop using it. :)

**Reply by sdottaka** [reply]
Initially, the default for "Include unique subfolders contents" was disabled.
The default has been changed to enabled because it is more problematic to not see items in one subfolder than to be slow.

**Reply by ICEknigh7** [reply]
How would that be problematic? Serious question because I can't think of a reason why that's not the default option, since the current default slows things down considerably without any benefit when there's nothing to be compared against.

**Reply by sdottaka** [reply]
> How would that be problematic? Serious question because I can't think of a reason why that's not the default option, since the current default slows things down considerably without any benefit when there's nothing to be compared against.

Showing contents of folders that exist only on one side helps avoid mistakes.
If such folders look empty, users might wrongly assume they were deleted, unimportant, or not worth copying.
Skipping them silently can lead to incorrect decisions, so showing them by default is safer.
